Multi Sources

fralken opened the pull request:
This is a PR related to issue #55.
Changes were made to the Source classes and to Device.

elliotwoods commented:
thanks for this!
i will definitely check this out when i have time
on a deadline right now. please hold :)
it's quite a big change so i want to look into it propely

fralken commented:
Well, from the user's point of view just one method has been added to the `Device` class, `initMultiSource` that can be invoked passing a list of FrameSourceTypes, like so:
```
	kinect.initMultiSource({
		FrameSourceTypes_Depth,
		FrameSourceTypes_Color,
		FrameSourceTypes_Body,
		FrameSourceTypes_BodyIndex });
```
As for the implementation, the idea is to split the update of Sources in two parts, so that update of the frame data is separated from the frame retrieval from the reader (so that the frame can be read from the MultiSourceReader instead). Same idea for the Source initialization, so that the Source's reader and Source's data are initilized separately, e.g. in case we use the MultiSourceReader, the specidic Source's reader is not initilized. 

To implement this I changed the hierarchy of Source classes like so:
```
Base
  |-> BaseFrame
          |-> Body
          |-> BaseImage
                   |-> Color
                   |-> BaseImageSimple
                            |-> BodyIndex
                            |-> Depth
                            |-> Infrared
                            |-> LongExposureInfrared
```
I.e. I added BaseFrame to the hierarchy, which implements, among other things, a generic `update `method that invoke specific `update`s of frame data of the derived classes.
`Device` class implements the correct initialization of readers and the invocation of the correct updates.

fralken commented:
Btw I also fixed `Depth::getDepthInColorFrameMapping` where a wrong buffer size was computed for the mapping data

elliotwoods commented:
pulled! thank you so much for making this implementation
